### PR TITLE
CRUD applied

### PR DESCRIPTION
**Pull request by miguel0577**
This is a pull request from the "crud-applied" branch to the "blog-created" branch (last created branch)

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -1,4 +1,49 @@
...
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @article = Article.find(params[:id])
+    @article.destroy
+
+    redirect_to root_path, status: :see_other
```
Comment (estebancordoba) [inline app/controllers/articles_controller.rb:42]: Where are we redirected?

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -1,4 +1,49 @@
...
+  end
+
+  private
+    def article_params
+      params.require(:article).permit(:title, :body)
```
Comment (estebancordoba) [inline app/controllers/articles_controller.rb:47]: What does permit do?

File: app/models/article.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Article < ApplicationRecord
+    validates :title, presence: true
+    validates :body, presence: true, length: { minimum: 10 }
```
Comment (estebancordoba) [inline app/models/article.rb:3]: What does presence do?

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Rails.application.routes.draw do
   root "articles#index"
 
-  get "/articles", to: "articles#index"
+  resources :articles
```
Comment (estebancordoba) [inline config/routes.rb:4]: What does resources do?

File: test/fixtures/articles.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (estebancordoba) [inline test/fixtures/articles.yml:1]: is this file necessary?

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (miguel0577) [reply app/controllers/articles_controller.rb:42]: We are redirected in the browser to the root path because it is the principal access point to the articles with the status code 303.

File: app/controllers/articles_controller.rb (hunk 2 above)
Comment (miguel0577) [reply app/controllers/articles_controller.rb:47]: As its name says, it allows access to the title and body parameters to filter them with the Rails feature Strong Parameters.

File: app/models/article.rb (hunk 3 above)
Comment (miguel0577) [reply app/models/article.rb:3]: Presence validates that a title value and a body value contain at least one character and are not empty.

File: config/routes.rb (hunk 4 above)
Comment (miguel0577) [reply config/routes.rb:4]: The resource maps all conventional paths to a collection of resources that work together to perform CRUD operations, in this case, the articles.

File: test/fixtures/articles.yml (hunk 5 above)
Comment (miguel0577) [reply test/fixtures/articles.yml:1]: The file articles.yml was generated with the model for the articles. It's only necessary if tests with the articles are required.

**Review by estebancordoba — APPROVED** [review]
